python/pyspark/sql/functions.py

```diff
@@ -345,6 +347,34 @@ def levenshtein(left, right):
 
 @ignore_unicode_prefix
 @since(1.5)
+def regexp_extract(str, pattern, idx):
+    """Extract a specific(idx) group identified by a java regex, from the specified string column.
+
+    >>> df0 = sqlContext.createDataFrame([('100-200',)], ['str'])
+    >>> df0.select(regexp_extract('str', '(\d+)-(\d+)', 1).alias('d')).collect()
```

Comment:
nit: according to a comment from @davies, there is no need to use `df0` instead of `df`. All tests get evaluated separately.